src/utils.ts

```diff
@@ -155,4 +155,9 @@ export default class Utils {
       target.assetQuery.templatedTarget
     }_${target.assetQuery.includeSubtrees}`;
   }
+
+  // used for generating the options.requestId
+  static getRequestId(options: QueryOptions, target: QueryTarget) {
+    return `${options.dashboardId}_${options.panelId}_${target.refId}`;
```

Comment:
Don't we need to encode the time ranges and stuff per query?

Reply:
Not for this, we are cancelling requests that come from the same target (ie each row for a graph). I think we only really need `panelId` and `targetRefId`, but I'm including the `dashboardId` just in case Grafana adds a way in the future to show multiple dashboards at once. 

The point of this is that if I send a bunch of expensive queries, for example if I'm under `Select Timeseries from Asset` and I'm clicking a bunch of checkboxes, each time I click one I'll be sending a new request and canceling the previous request. 
_(Currently, there is the potential for this: The user selects the `Select All` checkbox sending Request A, and then deselects a checkbox sending Request B immediately after. Now let's suppose Request B comes back first, Grafana updates the graph to go with Request B data. Then once Request A resolves and returns, Grafana will now update with Request A data. So we now have a graph that does not correspond with the query.)_

Reply:
But I did realize from your comment that I need to make a small change to the cache, since if I'm now encoding the requestId, we'll never use the results from a request for a duplicate target